src/create-dashboard.js

```diff
@@ -63,34 +72,36 @@ module.exports = function (selector, diJSON, visOpts) {
 
   // TODO: We can probably move this logic somewhere else
   var widgetModels = [];
-  diJSON.widgets.forEach(function (widget) {
-    var widgetAttrs = _.omit(widget, 'dataview');
+  vizJSON.widgets.forEach(function (rawWidgetData) {
+    var layerId = rawWidgetData.layerId;
+    var widgetAttrs = _.omit(rawWidgetData, 'options', 'layerId');
+    var dataviewModel;
 
     // Create dataview if there's a definition provided
-    var dataviewModel;
-    var d = widget.dataview;
-    if (d) {
+    var dataviewAttrs = rawWidgetData.options;
+    if (dataviewAttrs) {
+      // TODO not ideal, should have a more maintainable way of mapping
+      dataviewAttrs.type = rawWidgetData.type === 'time-series'
+        ? 'histogram' // Time-series widget is represented by a histogram, so re-map the type
+        : rawWidgetData.type;
+
       var layer;
 
       // Find the Layer that the Widget should be created for.
-      // a layerId has top-priority, otherwise it tries with a layerIndex, and even a subLayerIndex (if available)
-      if (d.layerId) {
-        layer = vis.interactiveLayers.find(function (m) {
-          return d.layerId === m.get('id');
+      if (layerId) {
+        layer = vis.map.layers.find(function (m) {
+          return layerId === m.get('id');
         });
-      } else if (Number.isInteger(d.layerIndex)) {
-        layer = vis.map.layers.at(d.layerIndex);
-        if (layer && Number.isInteger(d.subLayerIndex)) {
-          layer = layer.layers.at(d.subLayerIndex);
-        }
+      } else if (Number.isInteger(rawWidgetData.layerIndex)) {
+        // TODO Since namedmap doesn't have ids we need to map in another way, here using index
+        //   should we solve this in another way?
+        layer = vis.map.layers.at(rawWidgetData.layerIndex);
```

Comment:
@viddo I know why tried before, but I don't understand why it didn't worked now... As @rochoa says, there's no need for Windshaft to be aware of this ids. All we need is a way to get the layer by an attribute defined in the widget config (in the viz.json), so as long as the layer models in the the `map.layers` collection have been given the ids that are specified in the viz.json and widgets use those same `layerId`s, the following should work:


```
 layer = vis.map.layers.at(rawWidgetData.layerId);
```

Perhaps we should do a bit of debugging to see why it didn't work before.